inference_client/tasks/encode.py

```diff
@@ -0,0 +1,184 @@
+        """
+        ...
+
+    @overload
+    def encode(
+        self,
+        image: Union[
+            str,
+            bytes,
+            'ArrayType',
+            Iterable[str],
+            Iterable[bytes],
+            Iterable['ArrayType'],
+        ],
+        **kwargs,
+    ):
+        """
+        Encode image
+
+        :param image: the image to encode, can be a `ndarray`, 'bytes' or uri of the image
+        :param kwargs: additional arguments to pass to the model
+        """
+        ...
+
+    @overload
+    def encode(self, docs: Union[Iterable['Document'], 'DocumentArray'], **kwargs):
+        """
+        Encode documents
+
+        :param docs: the documents to encode
+        :param kwargs: additional arguments to pass to the model
+        """
+        ...
+
+    @overload
+    def encode(
+        self,
+        docs: Optional[Union[Iterable['Document'], 'DocumentArray']] = None,
+        text: Optional[Union[str, Iterable[str]]] = None,
+        image: Optional[
+            Union[
+                str,
+                bytes,
+                'ArrayType',
+                Iterable[str],
+                Iterable[bytes],
+                Iterable['ArrayType'],
+            ]
+        ] = None,
+        **kwargs,
+    ):
+        """
+        Encode text, image, or documents using a pre-trained model.
+
+        :param docs: the documents to encode. Default: None.
+        :param text: the text to encode. Default: None.
+        :param image: the image to encode, can be a `ndarray`, 'bytes' or uri of the image. Default: None.
+        :param kwargs: additional arguments to pass to the model.
+        """
+        ...
+
+    def encode(self, **kwargs):
+        """
+        Encode the documents using the model.
+
+        :param kwargs: additional arguments to pass to the model
+        :return: encoded content
+        """
+        payload, content_type, is_list = self._get_enocde_payload(**kwargs)
+        result = self.client.post(**payload)
+        return self._unbox_encode_result(
+            result=result,
+            content_type=content_type,
+            is_list=is_list,
+        )
+
+    def _get_enocde_payload(self, **kwargs):
+        payload = get_base_payload('/encode', self.token, **kwargs)
+        is_list = False
+
+        if 'docs' in kwargs:
+            if 'text' in kwargs or 'image' in kwargs:
+                raise ValueError(
+                    'More than one input type provided. Please provide only text, image or docs input.'
+                )
+            content_type = 'docarray'
+            total_docs = (
+                len(kwargs.get('docs'))
+                if hasattr(kwargs.get('docs'), '__len__')
+                else None
+            )
+            payload.update(total_docs=total_docs)
+            payload.update(inputs=iter_doc(kwargs.pop('docs')))
+
+        elif 'text' in kwargs:
+            if 'image' in kwargs:
+                raise ValueError(
```

Comment:
why not support?

Reply:
No available model yet.
We can waive this restriction once we have a multi-modality encode model

Reply:
BLIP2 support this, but not online now